beacon-chain/state/stategen/replayer.go

```diff
@@ -93,6 +93,10 @@ func (rs *stateReplayer) ReplayBlocks(ctx context.Context) (state.BeaconState, e
 		msg := fmt.Sprintf("error subtracting state.slot %d from replay target slot %d", s.Slot(), rs.target)
 		return nil, errors.Wrap(err, msg)
 	}
+	if diff == 0 {
+		return s, nil
+	}
```

Comment:
My main consideration with this PR would be to make sure that the ReplayToSlot still works as expected as it calls ReplayBlocks under the hood.

Reply:
It should work as expected given `s` is cached, checked, and returned here